### DISCUSSION TITLE
Livestream Links

### DISCUSSION THREAD

**Discussion by codereport** [codereport/jsource#6] [General]
Because YouTube doesn't allow sharing links in livestream chat, feel free to share them here and just ping me in the chat if you want me to check it out.

**Comment by codereport** [comment]
[Numb3rs Intro](https://www.youtube.com/watch?v=2zOSZmPtV30)

**Comment by codereport** [comment]
Links from Jan 17, 2021:
* [OhMyZsh](https://github.com/ohmyzsh/ohmyzsh)
* [23 Ranges: slide & stride - Conor Hoekstra - CppCon 2019](https://youtu.be/-_lqZJK2vjI)
* [CppCon 2019: Jason Turner “The Best Parts of C++"](https://youtu.be/iz5Qx18H6lg)
* [Thinking Forth](http://thinking-forth.sourceforge.net)
* [Dex - Google PL](https://openreview.net/pdf?id=rJxd7vsWPS)
* [Bosque - Microsoft PL](https://en.wikipedia.org/wiki/Bosque_(programming_language))
* [ShapeRank - F5 PL](https://github.com/f5devcentral/shapeRank)
* [Awesome D docs](https://dlang.org/library/std/algorithm/iteration/chunk_by.html)
* [Aaron Hsu's co-dfns](https://github.com/Co-dfns/Co-dfns)
* [Haskell Fantasy Birds](https://github.com/fantasyland/fantasy-birds)
* [To Mock a Mocking Bird](https://books.google.ca/books/about/To_Mock_a_Mockingbird.html?id=wklinRKAIgQC&printsec=frontcover&source=kp_read_button&redir_esc=y#v=onepage&q&f=false)
* [git lg](https://coderwall.com/p/euwpig/a-better-git-log)

**Comment by ahmadbeirkdar** [comment]
This may work. 

```
language: cpp
compiler:
  - clang
  - gcc
os:
  - linux
  - osx
before_install:
  - if [ "$TRAVIS_OS_NAME" = “linux” ]; then sudo apt-get -y install ninja-build; fi
  - if [ "$TRAVIS_OS_NAME" = “linux” ]; then pip install cmake; fi
  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew update; fi;
  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install cmake; fi;
  - if [ "$TRAVIS_OS_NAME" = "osx" ]; then brew install ninja-build; fi
script:
  - cmake --version; cmake --log-level=VERBOSE -G "Ninja Multi-Config" -B build; ninja -j12 -C build
branches:
  only:
    - main
 ```

**Comment by ahmadbeirkdar** [comment]
This may be a better way to install ninja, vs doing brew update and then brew install
https://docs.travis-ci.com/user/installing-dependencies/#installing-packages-on-macos

**Comment by ahmadbeirkdar** [comment]
```
language: cpp
compiler:
  - clang
  - gcc
os:
  - osx
addons:
  homebrew:
    packages:
    - ninja
script:
  - cmake --version; cmake --log-level=VERBOSE -G "Ninja Multi-Config" -B build; ninja -j12 -C build
branches:
  only:
    - main
 ```

**Reply by codereport** [reply]
Links from Livestream #3:
* [Meetup: Category Theory for Programmers](https://www.meetup.com/Programming-Languages-Toronto-Meetup/events/275918451/)
* [An Implementation of J](https://www.jsoftware.com/books/pdf/aioj.pdf)
* [Dyalog (APL) YouTube Channel](https://www.youtube.com/user/DyalogLtd/videos)
* [Thinking in APL](https://www.youtube.com/watch?v=myoK22rq1jk)
* [J Incunabulum](https://code.jsoftware.com/wiki/Essays/Incunabulum)
* [Buildkite](buildkite.com/)

**Comment by codereport** [comment]
Links from Livestream #4:

* https://en.cppreference.com/w/cpp/language/operator_precedence
* https://www.jsoftware.com/books/pdf/aioj.pdf
* https://dl.acm.org/doi/10.1145/79173.79184
* https://en.cppreference.com/w/cpp/utility/functional/boyer_moore_searcher
* https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines
* https://isocpp.org/wiki/faq/mixing-c-and-cpp

**Comment by unterumarmung** [comment]
oneTBB link to checkout
https://spec.oneapi.com/versions/latest/elements/oneTBB/source/nested-index.html

**Comment by codereport** [comment]
Point free programming talks:
* https://www.youtube.com/watch?v=ZS2_X-ZjFBI&t=1489s
* https://www.youtube.com/watch?v=6BnVo7EHO_8
* https://youtu.be/seVSlKazsNk

**Comment by joaogui1** [comment]
Flax has tables without the extra line: https://raw.githubusercontent.com/google/flax/master/docs/examples.rst

**Reply by joaogui1** [reply]
Ok, I'll explain my comment. Flax is a Deep Learning library based on JAX, what I'm linking is their raw documentation, which is rendered using sphinx and hosted here https://flax.readthedocs.io/en/latest/ which doesn't seem to need the extra line to render
Since you said you're interested in XLA here's a good playlist: https://www.youtube.com/playlist?list=PLqRUObJg9NTLwegKLg0CJikdibkVpAWq8
As for JAX there are many good links, if you want to I can post a couple of links here
